pkg/netpol/eval/resources.go

```diff
@@ -203,6 +209,41 @@ func (pe *PolicyEngine) upsertNamespace(ns *corev1.Namespace) error {
 	return nil
 }
 
+// checkLegalOwnerRefNameToPodLabels returns error if there are pod resources with same ownerReferences name but different labels
+func (pe *PolicyEngine) checkLegalOwnerRefNameToPodLabels(namespace, ownerRefName string, podLabels map[string]string) error {
+	if ownerRefName == "" { // pod does not have owner references
+		return nil
+	}
+	if _, ok := pe.ownersToLabelsMap[namespace]; !ok { // add the namespace to the map
+		pe.ownersToLabelsMap[namespace] = make(map[string]map[string]string)
+	}
+	ownerLabels, ok := pe.ownersToLabelsMap[namespace][ownerRefName]
+	if !ok { // add the new ownerReference with pod labels to the map
+		pe.ownersToLabelsMap[namespace][ownerRefName] = podLabels
+		return nil
+	}
+	// compare ownerLabels with podLabels
+	if !reflect.DeepEqual(ownerLabels, podLabels) {
+		return errors.New("Resources not supported for connectivity analysis. Pods with the ownerReferences' Name: " + ownerRefName +
+			" have different labels. Some labels' keys with different values: " + strings.Join(diffBetweenLabels(podLabels, ownerLabels), ","))
```

Comment:
In order to print these messages I need also to save the first pod name which its labels are saved in the new map as the ownerRef's labels.
Instead of using a new struct with pod name and its labels.
I changed the map to be from namespace to pods' ownerRef. name to its first pod.
then when comparing, I can compare curr. pod labels with first pod's labels and raise the relevant error message when we meet the first different pod